docs/api/serach.md

```diff
@@ -0,0 +1,1184 @@
+  - Search.Response
+    |field|type|label|desc.|
+    |:---:|:---|:---|:---|
+    |request_id|string| | the unique request ID |
+    |results|Object.Distance| repeated(Array[Object.Distance]) | search results |
+
+  - Object.Distance
+    |field|type|label|desc.|
+    |:---:|:---|:---|:---|
+    |id|string| | the vector ID |
+    |distance|float| | the distance between result vector and request vector |
+
+### Status Code
+
+| code | desc.            |
+| :--: | :--------------- |
+|  0   | OK               |
+|  3   | INVALID_ARGUMENT |
+|  5   | NOT_FOUND        |
+
+## LinearSearch RPC
+
+LinearSearch RPC is the method to linear search vector(s) similar to request vector.
+
+### Input
+
+- the scheme of `payload.v1.Search.Request`
+
+  ```bash
+  message Search {
+    message Request {
+      repeated float vector = 1 [ (validate.rules).repeated .min_items = 2 ];
+      Config config = 2;
+    }
+
+    message Config {
+      string request_id = 1;
+      uint32 num = 2 [ (validate.rules).uint32.gte = 1 ];
+      int64 timeout = 5;
+      Filter.Config ingress_filters = 6;
+      Filter.Config egress_filters = 7;
+    }
+  }
+  ```
+
+  - Search.Request
+    |field|type|label|required|desc.|
+    |:---:|:---|:---|:---:|:---|
+    |vector|float| repeated(Array[float]) | \* | the vector data. its dimension is between 2 and 65,536.|
+    |config|Config| | \* | the configuration of the search request |
+
+  - Search.Config
+    |field|type|label|required|desc.|
+    |:---:|:---|:---|:---:|:---|
+    |request_id|string| | | unique request ID |
+    |num|uint32| | \* | the maximum number of result to be returned |
+    |timeout|int64| | | Search timeout in nanoseconds (default value is `5s`) |
+    |ingress_filters|Filter.Config| | | Ingress Filter configuration |
+    |egress_filters|Filter.Config| | | Egress Filter configuration |
+
+### Output
+
+- the scheme of `payload.v1.Search.Response`.
+
+  ```bash
+  message Search {
+    message Response {
+      string request_id = 1;
+      repeated Object.Distance results = 2;
+    }
+  }
+
+  message Object {
+    message Distance {
+      string id = 1;
+      float distance = 2;
+    }
+  }
+  ```
+
+  - Search.Response
+    |field|type|label|desc.|
+    |:---:|:---|:---|:---|
+    |request_id|string| | the unique request ID |
+    |results|Object.Distance| repeated(Array[Object.Distance]) | search results |
+
+  - Object.Distance
+    |field|type|label|desc.|
+    |:---:|:---|:---|:---|
+    |id|string| | the vector ID |
+    |distance|float| | the distance between result vector and request vector |
+
+### Status Code
+
+| code | desc.            |
+| :--: | :--------------- |
+|  0   | OK               |
+|  3   | INVALID_ARGUMENT |
+|  5   | NOT_FOUND        |
+
+## LinearSearchByID RPC
+
+LinearSearchByID PRC it the method to linear search similar vectors using by user defined vector ID.<br>
+The vector whose ID is same to requested ID should be indexed into the `vald-agent` before searching.
```

Comment:
It think the not exist error will be returned if the ID is not exists in vald cluster, so I think it is not `should be`.
What do you think?

Reply:
yap, so I added `before searching`.
```suggestion
The vector whose ID is the same as the requested ID should be indexed into the `vald-agent` before searching.
If the vector doesn't be stored, you will get a `NOT_FOUND` error as a result.
```
how is it?